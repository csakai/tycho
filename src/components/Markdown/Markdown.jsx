import React from 'react';
import ReactMarkdown from 'react-markdown';
import PropTypes from 'prop-types';

export default class Markdown extends React.Component {

    static propTypes = {
        text: PropTypes.string.isRequired
    }

    render() {
        return (
            <div className="markdown">
                <ReactMarkdown source={this.props.text} />
            </div>
        )
    }
}