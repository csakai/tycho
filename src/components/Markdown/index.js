import './Markdown.css';
export {default} from './Markdown';