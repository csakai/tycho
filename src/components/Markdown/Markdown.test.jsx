import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Markdown from './Markdown';

describe('Markdown Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Markdown text="Hello, world!" />);
  });

  it('should render the markdown container successfully', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});