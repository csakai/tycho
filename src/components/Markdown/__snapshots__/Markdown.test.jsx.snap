// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Component should render the markdown container successfully 1`] = `
<div
  className="markdown"
>
  <ReactMarkdown
    containerTagName="div"
    parserOptions={Object {}}
    source="Hello, world!"
  />
</div>
`;
